s3/pyproject.toml

```diff
@@ -11,6 +11,8 @@ ops = ">=2.18.0"
 # TODO: Move into charm-libs groups after canonical/data-platform-libs#201 is merged
 pydantic = "^2.11.3"
 tenacity = "^9.1.2"
+boto3 = ">=1.39.4,<1.40"
```

Comment:
What is the reason to stay `<1.40`?

Reply:
No specific reason -- I have updated it to the latest version `1.40.17`.